spec/sec-regexp-regular-expression-objects-patch.html

```diff
@@ -246,181 +284,231 @@ <h1>GetStringIndex ( _S_, _Input_, _e_ )</h1>
     <emu-alg>
     1. Let _R_ be the *this* value.
     1. If Type(_R_) is not Object, throw a *TypeError* exception.
     1. Let _S_ be ? ToString(_string_).
+    1. <ins>Let _matchOptions_ be ? ToMatchOptions(_options_).</ins>
     1. Let _C_ be ? SpeciesConstructor(_R_, %RegExp%).
     1. Let _flags_ be ? ToString(? Get(_R_, `"flags"`)).
-    1. Let _matcher_ be ? Construct(_C_, &laquo; _R_, _flags_ &raquo;).
+    1. Let _matcher_ be ? Construct(_C_, « _R_, _flags_ »).
     1. Let _lastIndex_ be ? ToLength(? Get(_R_, `"lastIndex"`)).
     1. Perform ? Set(_matcher_, `"lastIndex"`, _lastIndex_, *true*).
     1. If _flags_ contains `"g"`, let _global_ be *true*.
     1. Else, let _global_ be *false*.
     1. If _flags_ contains `"u"`, let _fullUnicode_ be *true*.
     1. Else, let _fullUnicode_ be *false*.
-    1. <ins>If _mapfn_ is not *undefined* and IsCallable(_mapfn_) is *false*, throw a *TypeError* exception.</ins>
-    1. Return ! CreateRegExpStringIterator(_matcher_, _S_, _global_, _fullUnicode_<ins>, _mapfn_</ins>).
+    1. <del>Return ! CreateRegExpStringIterator(_matcher_, _S_, _global_, _fullUnicode_).</del>
+    1. <ins>Return ! CreateRegExpMatchIterator(_matcher_, _S_, _global_, _fullUnicode_, _matchOptions_).</ins>
     </emu-alg>
     <p>The value of the *name* property of this function is *"[Symbol.matchAll]"*.</p>
 
-  <emu-clause id="sec-createregexpstringiterator" aoid="CreateRegExpStringIterator">
-    <h1>CreateRegExpStringIterator ( _R_, _S_, _global_, _fullUnicode_<ins>, _mapfn_</ins> )</h1>
-    <p>The abstract operation _CreateRegExpStringIterator_ is used to create such iterator objects. It performs the following steps:</p>
-    <emu-alg>
-    1. Assert: Type(_S_) is String.
-    1. Assert: Type(_global_) is Boolean.
-    1. Assert: Type(_fullUnicode_) is Boolean.
-    1. <ins>Assert: _mapfn_ is *undefined* or IsCallable(_mapfn_) is *true*.</ins>
-    1. Let _iterator_ be ObjectCreate(<emu-xref href="#table-properties-of-regexp-string-iterator-instances">%RegExpStringIteratorPrototype%</emu-xref>, &laquo; [[IteratingRegExp]], [[IteratedString]], [[Global]], [[Unicode]]<ins>, [[Mapper]]</ins>, [[Done]] &raquo;).
-    1. Set _iterator_.[[IteratingRegExp]] to _R_.
-    1. Set _iterator_.[[IteratedString]] to _S_.
-    1. Set _iterator_.[[Global]] to _global_.
-    1. Set _iterator_.[[Unicode]] to _fullUnicode_.
-    1. <ins>Set _iterator_.[[Mapper]] to _mapfn_.</ins>
-    1. Set _iterator_.[[Done]] to *false*.
-    1. Return _iterator_.
-    </emu-alg>
+    <emu-clause id="sec-createregexpmatchiterator" aoid="CreateRegExpMatchIterator">
+      <h1><del>CreateRegExpStringIterator</del><ins>CreateRegExpMatchIterator</ins> ( _R_, _S_, _global_, _fullUnicode_<ins>, _matchOptions_</ins> )</h1>
+      <p>The abstract operation <del>_CreateRegExpStringIterator_</del><ins>_CreateRegExpMatchIterator_</ins> is used to create such iterator objects. It performs the following steps:</p>
```

Comment:
please revert this naming change; it is a string Iterator regardless of what values it produces (it’s named this not because it produces strings, but because it iterates them)